src/test/java/com/google/research/bleth/simulator/TracingSimulationBuilderTest.java

```diff
@@ -0,0 +1,239 @@
+package com.google.research.bleth.simulator;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TracingSimulationBuilderTest {
+
+    private static final IMovementStrategy MOVE_UP = new UpMovementStrategy();
+    private static final IMovementStrategy STATIONARY = new StationaryMovementStrategy();
+
+    private static final int BOARD_DIMENSION_EQUALS_TWO = 2;
+    private static final int MAX_ROUNDS_EQUALS_TWO = 2;
+    private static final int NUMBER_OF_BEACONS_EQUALS_TWO = 2;
+    private static final int NUMBER_OF_OBSERVERS_EQUALS_TWO = 2;
+    private static final double RADIUS_EQUALS_ONE = 1.0;
+    private static final int AWAKENESS_CYCLE_EQUALS_TWO = 2;
+    private static final int AWAKENESS_DURATION_EQUALS_ONE = 1;
+    private static final int AWAKENESS_DURATION_EQUALS_THREE = 3;
+    private static final AwakenessStrategyFactory.Type FIXES_AWAKENESS_STRATEGY_TYPE =
+            AwakenessStrategyFactory.Type.FIXED;
+
+    private static final int ILLEGAL_BOARD_DIMENSION = 0;
```

Comment:
If a constant is used only once, prefer to just inline it to the specific test that uses it. (You can still give it a name, e.g. as "`final int illegalBoardDimension = 0`"; it just ensures the definition is close to the use.)

[In general, prefer to create constants for test values only when (a) the constant is useful in multiple tests, and (b) nothing matters about the constant except what's in its name.]